### PR TITLE
PR - New Toolbar

### LINKED ISSUE #1901: New Toolbar
Remove toolbar in cycle landing page
(Please note, in screenshot do not include FRA 2020)
Image

Image

### PR DESCRIPTION
**Pull request by rfontanarosa**
close #1901 

https://user-images.githubusercontent.com/10047945/210850606-6ac8cf27-fc28-46fc-859e-e912e877ac8d.mov

https://user-images.githubusercontent.com/10047945/210850779-20f5e151-b6f7-4061-9992-87647f5f4c41.mov

Screenshot 2023-01-07 alle 17 41 52

### CODE REVIEW

File: src/client/components/PageLayout/Toolbar/Lock/Lock.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { useAppDispatch } from '@client/store'
+import { DataLockActions, useIsDataLocked } from '@client/store/ui/dataLock'
+import Icon from '@client/components/Icon'
+import { Breakpoints } from '@client/utils'
+
+const Lock: React.FC = () => {
+  const dispatch = useAppDispatch()
+
+  const locked = useIsDataLocked()
+  const canToggleLock = true // false // TODO
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Lock/Lock.tsx]: Is this todo still valid ?

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+import Lock from './Lock'
+import Status from './Status'
+import ToggleNavigationControl from './ToggleNavigationControl'
+
+const findElementRoot = (el: Element): Element => {
+  if (el.parentElement === null) return el
+  return findElementRoot(el.parentElement)
+}
+
+const CountrySelect: React.FC = () => {
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Toolbar.tsx]: Rename to Toolbar

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+              </>
+            )}
+          </div>
+          <Icon name="small-down" />
+
+          {open && <CountryList query={query} />}
+        </button>
+      </div>
+
+      {isFRA && isCountry && (
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Toolbar.tsx]: Remove isFra everywhere .

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        <MediaQuery minWidth={Breakpoints.laptop}>
+          {user && <Status />}
+
+          {user && country && isFRA && isCountry && <Lock />}
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Toolbar.tsx]: Lock should be rendered if user can edit data

File: src/client/components/Tooltip/Tooltip.tsx [hunk 5]
```diff
@@ -1,17 +1,19 @@
 import './Tooltip.scss'
 import React from 'react'
 
+import classNames from 'classnames'
+
 type Props = {
   text: string
-  children: any
+  children: JSX.Element
```
Comment (minotogna) [inline src/client/components/Tooltip/Tooltip.tsx]: Do not use children props. Use react with children type

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx (hunk 4 above)
Comment (rfontanarosa) [reply src/client/components/PageLayout/Toolbar/Toolbar.tsx]: canEditData is for sections, canEditCountryProps doesn't consider Collaborators.

we need another level of permission based on Cycle/Country/Role. Let's discuss about it.

File: src/client/components/PageLayout/Header/CycleSwitcher/CycleSwitcher.tsx [hunk 6]
```diff
@@ -23,9 +26,8 @@ const CycleSwitcher = () => {
   const userCycles = assessment.cycles.filter((cycle) => Authorizer.canView({ countryIso, user, cycle, assessment }))
   const canSwitchCycle = (isDataLocked || isDataExportView) && userCycles.length > 1
 
-  const onSelectChange = useCallback(
-    (event: React.ChangeEvent<HTMLSelectElement>) => {
-      const cycleName = event.target.value
+  const cycleChange = useCallback(
```
Comment (sorja) [inline src/client/components/PageLayout/Header/CycleSwitcher/CycleSwitcher.tsx]: onCycleChange

**Review by sorja — APPROVED** [review]

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx (hunk 4 above)
Comment (minotogna) [reply src/client/components/PageLayout/Toolbar/Toolbar.tsx]: Lock is for edit data. Why another level of permissions ?

File: src/client/components/PageLayout/Toolbar/Toolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+          {open && <CountryList query={query} />}
+        </button>
+      </div>
+
+      {isCountry && (
+        <MediaQuery minWidth={Breakpoints.laptop}>
+          {user && <Status />}
+
+          {user && country && <Lock />}
```
Comment (minotogna) [inline src/client/components/PageLayout/Toolbar/Toolbar.tsx]: Lock should be rendered if user can edit data

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by minotogna — APPROVED** [review]

**Pull request merged**